geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/AlterRegionCommandDUnitTest.java

```diff
@@ -55,8 +55,8 @@
   @ClassRule
   public static GfshCommandRule gfsh = new GfshCommandRule();
 
-  @Rule
-  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  @ClassRule
```

Comment:
This test is trying to use the `@Before` and `@After` to guarantee state between each test, when it should just be using the `@Rule` for the `GfshCommandRule` and `ClusterStartupRule`s.  Then again, it does seem to be about a minute of difference in runtime when I do them locally.

I'm torn between clean modularity between the tests and a pretty clear improvement on performance.  But while it's easy to say "Well, it's fine for *these* tests," it's hard to know if we're making a pitfall for future developers adding to this test class.

My gut tells me that using the `@Rule` for those would be the "right" thing to do, though.

Reply:
I don't think we have a rule to ensure that each tests start with no regions in the cache. It's just a way to ensure that each tests cleans after itself if the region is created in the test. I don't think it hurts to put a destroy in @After and an assert in @Before now that we have --if-exists flag. Having rules doesn't preclude the use of @Before and @After.